From ca96e5e5b7a92923d9fa7f26cb0d143590cb1ae7 Mon Sep 17 00:00:00 2001
Message-Id: <ca96e5e5b7a92923d9fa7f26cb0d143590cb1ae7.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Fri, 4 Jul 2014 22:43:37 +0530
Subject: [PATCH 079/312] FOR_UPSTREAM [VPG]: drm/i915: Fix the pipe B sprite
 DL program

In case of the sprite C/D we are using the pipe A dotclock instead of the pipe B
dot clock to program the drain latency.

Issue: APDEV-1171
OTC-Tracker: VIZ-3516
Change-Id: Ie9c914e29aaffe53e57d9ec4957bf5b4cf53afd6
Signed-off-by: Pallavi <pallavi.g@intel.com>
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Reviewed-on: http://android.intel.com:8080/124184
Reviewed-by: Kumar, Shobhit <shobhit.kumar@intel.com>
Reviewed-by: Noziska, Patrick J <patrick.j.noziska@intel.com>
Reviewed-by: Yang, Fei <fei.yang@intel.com>
Tested-by: Sun, Hang L <hang.l.sun@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
(cherry picked from commit 307fc199b67f7e943efd852849c3a01071b29f4e)

Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 6b7a108..10fb213 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3091,7 +3091,7 @@ static void valleyview_update_sprite_wm(struct drm_plane *plane,
 	enable.EnSprite = is_sprite_enabled(dev_priv, 1, 0);
 
 	val = I915_READ(VLV_DDL2);
-	if (vlv_compute_drain_latency(dev, 0, NULL, NULL, NULL, NULL,
+	if (vlv_compute_drain_latency(dev, 1, NULL, NULL, NULL, NULL,
 		&sprite_prec_mult, &sprite_dl, pixel_size, enable)) {
 		sprite_prec = (sprite_prec_mult ==
 				DRAIN_LATENCY_PRECISION_32) ?
@@ -3109,7 +3109,7 @@ static void valleyview_update_sprite_wm(struct drm_plane *plane,
 	enable.EnSprite = is_sprite_enabled(dev_priv, 1, 1);
 
 	val = I915_READ(VLV_DDL2);
-	if (vlv_compute_drain_latency(dev, 0, NULL, NULL, NULL, NULL,
+	if (vlv_compute_drain_latency(dev, 1, NULL, NULL, NULL, NULL,
 		&sprite_prec_mult, &sprite_dl, pixel_size, enable)) {
 		sprite_prec = (sprite_prec_mult ==
 				DRAIN_LATENCY_PRECISION_32) ?
-- 
1.7.9.5

